wemake_python_styleguide/visitors/tokenize/primitives.py

```diff
@@ -30,12 +31,35 @@ class WrongNumberTokenVisitor(BaseTokenVisitor):
         'X', 'O', 'B', 'E',
     ))
 
+    _bad_number_prefixes = ('0b0', '0x0', '0o0', 'e0')
+
     def _check_underscored_number(self, token: tokenize.TokenInfo) -> None:
         if '_' in token.string:
             self.add_violation(
                 UnderscoredNumberViolation(token, text=token.string),
             )
 
+    def _is2n_digits(self, token_length: int) -> bool:
+        # checking that literals have 2**n digits
+        return token_length & (token_length - 1) == 0
+
+    def _check_meaningless_zeros(self, token: tokenize.TokenInfo) -> None:
+        token_string = token.string.lower()
+        token_start_pos = token_string.find('e')
+        token_start_pos = token_start_pos if token_start_pos >= 0 else 0
+        for prefix in self._bad_number_prefixes:
```

Comment:
I think that the logic is too complicated. Maybe it will be better if prefix 'e0' (it's not even a prefix!) will be handled separately? Then you won't need to use this awkward `token_start_pos` for other prefixes and check for `!= 'e0'`.

Reply:
👍